ledger/src/blockstore.rs

```diff
@@ -2878,8 +2878,6 @@ impl Blockstore {
         slot: Slot,
         start_index: u64,
     ) -> Result<(CompletedRanges, Option<SlotMeta>)> {
-        let _lock = self.check_lowest_cleanup_slot(slot)?;
-
```

Comment:
I initially started upon a more complicated change but landed here. My reasoning for simply ripping this out is:
- The lock is only held for `get_completed_ranges()` so it wasn't providing us any guarantees for when we subsequently read the shreds from rocksdb
- This function issues a single (atomic) rocksdb read operation, so lock not needed for this function

So, by removing this, the only thing that we lose is distinguishing a slot being unavailable vs. the slot having been cleaned up when calling `get_completed_ranges()` directly. The only spot we care about `BlockstoreError::SlotCleanedUp` is in RPC after calling `get_rooted_block()`. But, since `get_rooted_block()` calls `check_lowest_cleanup_slot()` to check the slot and hold the lock, we don't have any change of behavior from RPC's point of view.

https://github.com/solana-labs/solana/blob/375f9ae41d80fb7ab93c13ee834388fe34c3786d/rpc/src/rpc.rs#L1017-L1038